sktime/utils/estimator_checks.py

```diff
@@ -1,8 +1,43 @@
 # -*- coding: utf-8 -*-
 """Estimator checker for extension."""
 
-__author__ = ["fkiraly"]
-__all__ = ["check_estimator"]
+__author__ = ["fkiraly", "miraep8"]
+__all__ = ["check_estimator", "delegate_if_needed"]
+
+
+def delegate_if_needed(func):
+    """Check that func is supported by the delegated estimator type.
+
+    Intended to be used as a decorator.
+    Will check whether func is supported for estimators of type(self.delegate_name)
+        if so - will delegate the call, else will raise error.
+
+    Parameters
+    ----------
+    func - the function in question - will be one of the decorated functions below
+
+    Raises
+    ------
+    TypeError if the func is not included in the list of methods supported by
+        self._delegate_name
+    """
+    from copy import deepcopy
+
+    def inner(*args, **kwargs):
+        self = args[0]
+        estimator = self._get_delegate()
+        valid_calls = dir(estimator)
+        if func.__name__ in valid_calls:
+            new_args = list(deepcopy(args))
+            new_args[0] = estimator
+            return func(*tuple(new_args), **kwargs)
```

Comment:
There is actually a "hidden" call to `getattr` in `self._get_delegate()` (which I borrowed from your original `DelegateForecaster`!).  I *think* what should be happening here is that we are swapping a reference to `self` for a reference to `estimator`, which should allow us to update `estimator` properly.  (The other reason why I believe `estimator` is in fact properly updated is that if it wasn't more tests would break... (ie, a call to `fit()` and then `predict()` might not work because `estimator`'s `_is_fitted` property never got properly updated)).  But let me know if you disagree/have further concerns.